### DISCUSSION TITLE
It doesn't work blabla.

### DISCUSSION THREAD

**Discussion by Bobsoftru** [AvaloniaUI/Avalonia#17567] [General]
I make a public class blabla: TextBox is not displayed on the screen, I change public class blabla: MaskedTextBox everything works as it should.

**Comment by thevortexcloud** [comment]
You need to share code examples for this.

**Reply by Bobsoftru** [reply]
Attach this to any window
   `public class Textmask : TextBox { public Textmask() { } }`

**Reply by thevortexcloud** [reply]
Just checking if you have tried setting a style key?

https://docs.avaloniaui.net/docs/basics/user-interface/styling/styles#style-key

**Reply by Bobsoftru** [reply]
Thanks, I'll look into it, but MaskedTextBox works.

**Reply by thevortexcloud** [reply]
That's probably because it sets a StyleKey:

https://github.com/AvaloniaUI/Avalonia/blob/28c21078f51ef3ad557801067d3fd72679a9b09a/src/Avalonia.Controls/MaskedTextBox.cs#L184

**Comment by Bobsoftru** [comment]
Can you give me an example, I'm still at the beginning of Avalonia ?

**Comment by Bobsoftru** [comment]
Everything worked, thanks
thevortexcloud, I don't see how to give you an answer.

**Reply by thevortexcloud** [reply]
Glad I can help. There should be a big "Mark as answer" button under the comment.

https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer

**Reply by Bobsoftru** [reply]
![image](https://github.com/user-attachments/assets/7cabbc95-8686-4d24-b92e-ddcc9a245a2f)
Sorry, I wasn't given such a button.

**Reply by rabbitism** [reply]
> ![image](https://github.com/user-attachments/assets/7cabbc95-8686-4d24-b92e-ddcc9a245a2f)
> Sorry, I wasn't given such a button.

That's only for discussion in Q＆A type, this one is General

**Reply by Bobsoftru** [reply]
I'll keep that in mind, thevortexcloud, my thanks.
